src/main/java/org/commonjava/service/promote/core/PromotionManager.java

```diff
@@ -442,16 +462,15 @@ private boolean reDownloadMissing(StoreKey storeKey, Set<String> paths)
         if (!isSuccess(resp))
         {
             logger.warn("Re-download existence check failed, status: {}", resp.getStatus() );
-            return false;
+            return emptySet(); // throw exception? maybe just ignore it and let promotion move forward
```

Comment:
Is there any impact if ignoring here ?

Reply:
If ignoring, it will take it like nothing missing and go ahead, then the promote will fail. but actually we can fail it earlier here, because the checking failure may indicate that the storage service has problem. I will modify it.

Reply:
done. pls review again.